File: kotlin/kotlin-vs-java/java-example/src/main/java/org/example/ex01/Exam01.java
package org.example.ex01;

/**
 * 변수 선언
 */
public class Exam01 {

    public Exam01() {
        int age = 10;
        Integer _age = 20;

        double d = 10d;
        Double _d = 20d;

        float f = 20f;
        Float _f = 20f;

        long l = 10L;
        Long _l = 10L;

        boolean b = true;
        Boolean _b = true;

        String name = "홍길동";

        System.out.printf("사용자 이름: %s", name);
    }

    public static void main(String[] args) {
        new Exam01();
    }
}

File: kotlin/kotlin-vs-java/kotlin-example/src/main/kotlin/ex01/Exam01.kt
package ex01

/**
 * 변수 선언
 */
fun main() {
    // var => mutable, val => immutable (=final)
    // 타입 생략 가능

    // primitive 타입 없음
    val i = 10;
    val _i: Int = 10;

    var d: Double = 20.0;
    var _d: Double;

    var f: Float = 10f;

    var name: String = "홍길동";
    val _name: String = "홍길동";
    val n = "홍길동";

    println("사용자 이름: $name");
}